doc/amoc_coordinator.md

```diff
@@ -0,0 +1,170 @@
+CoordinationActions :: [ CoordinationActions ] | CoordinationAction
+CoordinationAction :: 
+    fun(Event) -> any() |
+    fun(Event, ListOfUsersData) -> any() |
+    fun(Event, MaybeUser, MaybeUser) -> any()
+Event :: coordinate | timeout | stop | reset
+ListOfUsersData :: [ User ]
+MaybeUser :: undefined | User
+User :: {UsersPid :: pid(), Data :: any()}
+Timeout :: pos_integer() | infinity
+```
+
+This function starts a coordinator. Usually is called in `init/0` of a amoc scenario.
+
+#### `add(CoordinatorName, Data) -> ok`
+#### `add(CoordinatorName, Pid, Data) -> ok`
+
+##### Types
+```erlang
+CoordinatorName :: atom()
+UsersPid :: pid()
+Data :: any()
+```
+
+This function adds current process data. Usually is called in `start/2` of a amoc scenario.
+
+## Example
+
+This scenario will demonstrate how do the `users` interact with `amoc_coordinator`:
+
+```erlang
+-module(example).
+
+-export([init/0]).
+-export([start/2]).
+
+init() ->
+    Plan = [
+        {2, fun(Event) ->
+                io:fwrite("Two new users showed up: Event = ~p\n", [Event])
+            end},
+        {2, fun(Event, ListOfUsers) ->
+                io:fwrite("Two new users showed up: Event = ~p; ListOfUsers = ~p\n", [Event, ListOfUsers]),
+                [ Pid ! {hello, Data} || {Pid, Data} <- ListOfUsers]
+            end},
+        {2, fun(Event, User1, User2) ->
+                io:fwrite("Two new users showed up: Event = ~p; User1 = ~p; User2 = ~p\n", [Event, User1, User2])
+            end},
+        
+        {3, fun(_Event) ->
+                io:fwrite("Three new users showed up\n", [])
+            end},
+        {all, fun(Event) ->
+                io:fwrite("All users have called amoc_coordinator:add in Event = ~p\n", [Event])
+            end}
+        
+    ],
+    Settings = [setting1, {setting2, something}],
+    amoc_coordinator:start(?MODULE, Plan),
+    {ok, Settings}.
+
+start(Id, _Settings) ->
+    io:fwrite("User = ~p\n", [Id]),
+    amoc_coordinator:add(?MODULE, Id),
+    receive
+        Msg -> 
+            io:fwrite("{Msg = ~p, Id = ~p\n", [Msg, Id])
+    end,
+    ok.
+```
+
+
+To run it:
+
+```bash
+$ make rel
+$ _build/default/rel/amoc/bin/amoc console
+
+1> amoc_local:do(example, 1, 5).
+```
+
+Filtered, formated and explained output:
+
+```erlang
+User = 1  % First user is started
+
+ok  % result of calling amoc_local:do/3
+
+User = 2 % First user is started
+
+% We have 2 users added to amoc_coordinator so all of actions {2, _} are triggered:
+Two new users showed up: Event = coordinate; User1 = {<0.1142.0>,2}; User2 = {<0.1140.0>,1}
+% This action triggers sending {hello,Id} to the users 1 and 2
+Two new users showed up: Event = coordinate; ListOfUsers = [{<0.1142.0>,2},{<0.1140.0>,1}]
+Two new users showed up: Event = coordinate
+
+% Users 1 and 2 received messages and print them:
+{Msg = {hello,2}, Id = 2
+{Msg = {hello,1}, Id = 1
+
+User = 3
+% We have 3 users added to amoc_coordinator so all of actions {3, _} are triggered:
```

Comment:
> so all of the `{3, _}` actions [...]

(ordering of the english genitive, and one article)
(same in the sentence a few lines below)